jerry-ext/jerry-context/jerry-context.c

```diff
@@ -0,0 +1,139 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+#include "jerry-context.h"
+#include "jmem.h"
```

Comment:
No. That would be too much restriction. To me extensions are core part of the project, and those who write extensions understand jerryscript internals (and fix their extensions after breaking changes). The dependency is one way, so extensions can use core functions, but core functions cannot use extensions.

User applications should only use public jerry and jerry extension api.